lib/Conversion/TritonGPUToLLVM/ConvertLayoutOpToLLVM.cpp

```diff
@@ -442,50 +473,65 @@ struct ConvertLayoutOpConversion
     // unsigned elems = getTotalElemsPerThread(srcTy);
     auto vals = getTypeConverter()->unpackLLElements(loc, adaptor.getSrc(),
                                                      rewriter, srcTy);
-    unsigned inVec = 0;
-    unsigned outVec = 0;
-    auto paddedRepShape = getScratchConfigForCvtLayout(op, inVec, outVec);
 
     unsigned outElems = getTotalElemsPerThread(dstTy);
     auto outOrd = getOrder(dstLayout);
     SmallVector<Value> outVals(outElems);
 
+    unsigned inVec = 0;
+    unsigned outVec = 0;
+    SmallVector<unsigned> repShape;
+    triton::gpu::SharedEncodingAttr sharedLayout;
+    if (srcLayout.isa<BlockedEncodingAttr>() &&
+        dstLayout.isa<MmaEncodingAttr>() && !isDstMmaV1) {
+      repShape = getScratchConfigForCvtLayout(op, inVec, outVec, false);
+      SmallVector<int64_t> repShapeLong;
+      for (unsigned i = 0; i < repShape.size(); ++i) {
+        repShapeLong.push_back(repShape[i]);
+      }
+      auto dstDotOp = triton::gpu::DotOperandEncodingAttr::get(
+          getContext(), 0, dstLayout, dstTy.getElementType());
+      sharedLayout = triton::gpu::SharedEncodingAttr::get(
+          getContext(), dstDotOp, repShapeLong, getOrder(dstLayout),
+          srcTy.getElementType());
+    } else if (isSrcMmaV1 || isDstMmaV1) {
+      repShape = getScratchConfigForCvtLayout(op, inVec, outVec, true);
+      auto maxVec = std::max(inVec, outVec);
+      sharedLayout = triton::gpu::SharedEncodingAttr::get(
+          getContext(), maxVec, 1, repShape[0], getOrder(dstLayout));
```

Comment:
This should replicate the old code logic, when either src/dst is mma v1, but not sure if it actually does. `repShape` is padded, but we are still going to swizzle, maybe maxPhase should be 1 instead of repShape[0]